docs/contributor/adr/0007-compatibilityMode.md

```diff
@@ -0,0 +1,52 @@
+# Support for **compatibilityMode** Configuration
+
+## Status
+
+Accepted by the Istio module team
+
+## Context
+
+With Istio's introduction of a [compatibilityVersion](https://istio.io/latest/news/releases/1.21.x/announcing-1.21/#easing-upgrades-with-compatibility-versions) and environment variables allowing to mitigate breaking changes during Istio upgrade, Istio module team decided to introduce compatibility mode to allow users leverage possibility of remaining compatible with previous Istio versions.
+
+## Considerations
+
+Should `compatibilityVersion` be used, exposed by Istio repository with an additional possibility to add environment variables, or a custom implementation allowing Istio team to control the scope of the compatibility settings, either in the environment variables or other parts of the IstioOperator config?
+
+## Decision
+
+To avoid a situation, in which compatibilityVersion introduces changes that have breaking, or unwanted impact on the Istio configuration, Istio team decide not to use `compatibilityVersion`, and to create custom implementatation.
```

Comment:
I'm missing that this decision was mainly done, because compatibility version would again introduce configuration that was on purpose already removed from the Istio operator in module before.